third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -78,16 +97,18 @@ void dds_device::impl::set_state( state_t new_state )
     {
         if( _metadata_reader )
         {
-            if( rsutils::json::get( _participant->settings(), "disable-metadata", false ) )
+            nlohmann::json md_settings = rsutils::json::nested( _device_settings, "metadata" );
+            if( ! md_settings.is_null() && ! md_settings.is_object() )  // not found is null
             {
-                LOG_DEBUG( "... metadata is available but 'disable-metadata' is set" );
+                LOG_DEBUG( "... metadata is available but device/metadata is disabled" );
                 _metadata_reader.reset();
             }
             else
             {
                 LOG_DEBUG( "... metadata is enabled" );
                 dds_topic_reader::qos rqos( eprosima::fastdds::dds::BEST_EFFORT_RELIABILITY_QOS );
                 rqos.history().depth = 10; // Support receive metadata from multiple streams
+                rqos.from_json( md_settings );
```

Comment:
`from_json` is not clear because object is already created. `override_from_json` is better. Same for writer QoS

Reply:
Changed.